### PR TITLE
Avoid duplication query for isOwnerOfTeam && isOwner method

### PR DESCRIPTION
**Pull request by alikhosravidev**
This changes remove duplicate query when check user is owner of team

### CODE REVIEW

File: src/Traits/UserHasTeams.php [hunk 1]
```diff
@@ -73,7 +73,9 @@ public static function bootUserHasTeams()
      */
     public function isOwner()
     {
-        return ($this->teams()->where('owner_id', '=', $this->getKey())->first()) ? true : false;
+        return (bool) $this->teams
+            ->where('owner_id', '=', $this->getKey())
+            ->first();
```
Comment (okaufmann) [inline src/Traits/UserHasTeams.php:82]:
```suggestion
        return $this->teams
            ->where('owner_id', '=', $this->getKey())
            ->exists();
```

File: src/Traits/UserHasTeams.php [hunk 2]
```diff
@@ -112,10 +114,9 @@ public function isOwnerOfTeam($team)
     {
         $team_id = $this->retrieveTeamId($team);
 
-        return ($this->teams()
+        return (bool) $this->teams
```
Comment (okaufmann) [inline src/Traits/UserHasTeams.php:121]:
```suggestion
        return $this->teams
```

File: src/Traits/UserHasTeams.php [hunk 3]
```diff
@@ -112,10 +114,9 @@ public function isOwnerOfTeam($team)
...
             ->where('owner_id', $this->getKey())
-            ->where('team_id', $team_id)->first()
-        ) ? true : false;
+            ->where('id', $team_id)->first();
```
Comment (okaufmann) [inline src/Traits/UserHasTeams.php:123]:
```suggestion
            ->where('id', $team_id)->exists();
```

**Review by okaufmann — CHANGES_REQUESTED** [review]
I've added some suggestions

File: src/Traits/UserHasTeams.php (hunk 3 above)
Comment (ericbarrionuevo) [reply src/Traits/UserHasTeams.php:123]: hi this suggestions is very recommendable

Comment (alikhosravidev) [reply src/Traits/UserHasTeams.php:123]: thank you for suggestions

Comment (alikhosravidev) [reply src/Traits/UserHasTeams.php:123]: But your suggestion is not suitable for this reason:

Method Illuminate\Database\Eloquent\Collection::exists does not exist.

$this->teams return a collection object.

**Pull request closed without merging**